addons/subscription/models/subscription.py

```diff
@@ -0,0 +1,132 @@
+    exec_init = fields.Integer(string='Number of Documents')
+    date_init = fields.Datetime(string='First Date', default=fields.Datetime.now)
+    state = fields.Selection([('draft', 'Draft'), ('running', 'Running'), ('done', 'Done')], string='Status', copy=False, default='draft')
+    doc_source = fields.Reference(selection=_get_document_types, string='Source Document', required=True, help="User can choose the source document on which he wants to create documents")
+    doc_lines = fields.One2many('subscription.subscription.history', 'subscription_id', string='Documents created', readonly=True)
+    cron_id = fields.Many2one('ir.cron', string='Cron Job', help="Scheduler which runs on subscription", states={'running': [('readonly', True)], 'done': [('readonly', True)]})
+    note = fields.Text(string='Notes', help="Description or Summary of Subscription")
+
+    @api.model
+    def _auto_end(self):
+        super(Subscription, self)._auto_end()
+        # drop the FK from subscription to ir.cron, as it would cause deadlocks
+        # during cron job execution. When model_copy() tries to write() on the subscription,
+        # it has to wait for an ExclusiveLock on the cron job record, but the latter
+        # is locked by the cron system for the duration of the job!
+        # FIXME: the subscription module should be reviewed to simplify the scheduling process
+        #        and to use a unique cron job for all subscriptions, so that it never needs to
+        #        be updated during its execution.
+        self.env.cr.execute("ALTER TABLE %s DROP CONSTRAINT %s" % (self._table, '%s_cron_id_fkey' % self._table))
+
+    @api.multi
+    def set_process(self):
+        for subscription in self:
+            cron_data = {
+                'name': subscription.name,
+                'interval_number': subscription.interval_number,
+                'interval_type': subscription.interval_type,
+                'numbercall': subscription.exec_init,
+                'nextcall': subscription.date_init,
+                'model': self._name,
+                'args': repr([[subscription.id]]),
+                'function': 'model_copy',
+                'priority': 6,
+                'user_id': subscription.user_id.id
+            }
+            cron = self.env['ir.cron'].create(cron_data)
+            subscription.write({'cron_id': cron.id, 'state': 'running'})
+
+    @api.multi
+    def model_copy(self):
+        for subscription in self.filtered(lambda sub: sub.cron_id):
+            if not subscription.doc_source.exists():
+                raise UserError(_('Please provide another source document.\nThis one does not exist!'))
+
+            docsource_data = {'state': 'draft'}
+
+            documents = self.env['subscription.document'].search([('model.model', '=', subscription.doc_source._name)], limit=1)
+            datefields = documents.field_ids.filtered(lambda field: field.value == 'date')
```

Comment:
@dek-odoo @hmo-odoo  this is a __change of behaviour!__
Before:
```python
for f in doc.field_ids:
    if f.value=='date':
        value = time.strftime('%Y-%m-%d')
    else:
        value = False
    default[f.field.name] = value
```
Which may give you something like `default = {'name': False, 'date1': '2016-01-20', 'foo': False}`

With your code:
```python
documents.field_ids.filtered(lambda field: field.value == 'date')
fieldnames = datefields.mapped('field').mapped('name')
docsource_data.update(dict(zip(fieldnames, [fields.Date.today(self)] * len(fieldnames))))
```
that will gives `default = {'date1': '2016-01-20'}`

Using `filtered` and `map` is efficient but be sure to use it correctly. If you are not sure, just use a list comprehension, if we are working on small sets of records, that will not have any effect anyway.